x/bank/keeper/send.go

```diff
@@ -232,18 +232,30 @@ func (k BaseSendKeeper) subUnlockedCoins(ctx sdk.Context, addr sdk.AccAddress, a
 
 	for _, coin := range amt {
 		balance := k.GetBalance(ctx, addr, coin.Denom)
-		locked := sdk.NewCoin(coin.Denom, lockedCoins.AmountOf(coin.Denom))
-		spendable := balance.Sub(locked)
 
-		_, hasNeg := sdk.Coins{spendable}.SafeSub(coin)
+		// NOTE: denom and amount validation should occur before transfer
+		locked := sdk.Coin{
+			Denom:  coin.Denom,
+			Amount: lockedCoins.AmountOfNoDenomValidation(coin.Denom),
```

Comment:
What benefit do we have to calling `AmountOfNoDenomValidation` here over `AmountOf`? Since validation should occur prior as you pointed out, `AmountOf` seems practical, no?

Reply:
`NewCoin` and `AmountOf` validate the denom. Here we don't need to because we are already passing a `Coins` type as an argument

Reply:
I'm a bit against this. The idea is that these APIs exist for a reason, and unless explicitly not needed, e.g. denom validation is explicitly not required, then `New*` constructors should always be called. This also keeps consistency with the rest of the codebase and makes the PR diff smaller.

Reply:
I tend to agree on this one, I really like consistency. Unless there's a noticeable performance difference I would go with `AmountOf`.

Reply:
Shall we just push to this branch and fix it then?

Reply:
Yah!

Reply:
Reverted!